packages/shell-chrome/src/backend.js

```diff
@@ -121,113 +26,230 @@ function serializeDataProperty(value) {
     }
 }
 
-function discoverComponents() {
-    const rootEls = document.querySelectorAll('[x-data]')
-    // Exit early if no components have been added or removed
-    const allComponentsInitialized = Object.values(rootEls).every((e) => e.__alpineDevtool)
-    if (window.__alpineDevtool.components.length === rootEls.length && allComponentsInitialized) {
-        return false
+class AlpineDevtoolsBackend {
+    constructor() {
+        this.components = []
+        this.uuid = 1
+        this.hoverElement = null
+        this.observer = null
+        this._stopMutationObserver = false
```

Comment:
I assume we're using `_` prefix here to indicate a private field?

Reply:
yep, the mutation stuff is now all handled inside of the class